code/clue.js

```diff
@@ -4,56 +4,191 @@ const mrGreen = {
+  missScarlet,
+  mrsPeacock,
+  cMustard
 ]
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol,
+]
 
-const rooms = []
+const rooms = [
+  'Dining Room',
+  'Conservatory',
+  'Kitchen',
+  'Study',
+  'Library',
+  'Biliard Room',
+  'Lounge',
+  'Ballroom',
+  'Hall',
+  'Spa',
+  'Living Room',
+  'Observatory',
+  'Theater',
+  'GuestHouse',
+  'Patio',
+]
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
 // YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
-
+const randomSelector = arrays => {
+  return arrays[Math.floor(Math.random() * arrays.length)]
+} 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
 // With a killer, a weapon and a room.
 // The values will be set later.
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null,
+};
 
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
   // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
+  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. 
+  document.getElementById('killerCard').style.background = mystery.killer.color;
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+};
+document.getElementById("killerCard").onclick = pickKiller;
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms);
+
+  // document.getElementById('roomCard').style.background = mystery.room.color;
+  document.getElementById('roomName').innerHTML = `${mystery.room}`;
+};
+document.getElementById("roomCard").onclick = pickRoom;
+
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons);
+
+  document.getElementById('weaponCard').style.background = mystery.weapon.color;
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`;
+};
+document.getElementById("weaponCard").onclick = pickWeapon;
+
 
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+
+
+function revealMystery () {
```

Comment:
This functionality is still not fixed - the difference is that when we click "Reveal the crime" button now, we got error in Console tab.
I know you have far more important tasks to focus on right now Claudia, so there is a tip for you how to solve this issue : you need to prevent function from being called until all cards (killer, weapon, room) are selected. How can we do that? We know that when all cards are selected, both `mystery.killer`, `mystery.weapon` and `mystery.room` are not equal to `null`.